### DISCUSSION TITLE
What is best way to run docksal in gitlab pipeline

### DISCUSSION THREAD

**Discussion by msankhala** [docksal/docksal#1579] [Q&A] [answered]
I am trying to run docksal in gitlab pipeline. I am using ubuntu image to run my pipeline. On each run of the pipeline, I need to install docksal and docker, which is taking a lot of time. Is there any better approach to run docksal in the pipeline, Or I am completely missing it and docksal is not designed to run in the pipeline?

**Comment by lmakarov** [comment, marked as answer]
Take a look at these two repos:

- https://github.com/docksal/sandbox-server
- https://github.com/docksal/ci-agent

This is the officially supported approach for running Docksal in CI pipelines to build preview (sandbox) environments.

This video explains how it all works together: https://www.youtube.com/watch?v=6omXTGfS7gw
